src/Microsoft.Identity.Web/TokenCacheProviders/InMemory/MsalMemoryTokenCacheProvider.cs

```diff
@@ -92,6 +92,10 @@ protected override Task WriteCacheBytesAsync(string cacheKey, byte[] bytes)
             if (cacheSerializerHints != null && cacheSerializerHints?.SuggestedCacheExpiry != null)
             {
                 cacheExpiry = cacheSerializerHints.SuggestedCacheExpiry.Value.UtcDateTime - DateTime.UtcNow;
+                if (cacheExpiry < TimeSpan.Zero)
+                {
+                    cacheExpiry = TimeSpan.FromMilliseconds(1);
+                }
```

Comment:
Don't we want to add logging here too?

Reply:
we don't already have it here, so didn't want to add it just for this.